importers/importer.py

```diff
@@ -38,25 +49,30 @@ def get_id_of(table, name):
 
 def import_point(points):
     for point in points:
+        point.building_name = point.building_name or "UnID'd Building"
         building_id = get_id_of("buildings", point.building_name)
         if building_id is None:
-            execute_and_commit("""INSERT INTO buildings (name) VALUES (%s)""", (point.building_name,))
+            execute_and_commit("""INSERT INTO buildings (name) VALUES (%s)""",
+                               (point.building_name,))
             building_id = get_id_of("buildings", point.building_name)
 
         # use somewhere: {'floor': point.room_floor, 'building_id': building_id}
-        room_id = get_id_of("rooms", point.room_name)
+        point.room_name = point.room_name or "UnID'd Room in " + point.building_name
+        room_id = get_id_of("rooms", point.room_name, building_id=building_id)
         if room_id is None:
             execute_and_commit(
                 """INSERT INTO rooms (name, building_id, floor) VALUES (%s, %s, %s)""",
                 (point.room_name, building_id, try_cast_int(point.room_floor)))
-            room_id = get_id_of("rooms", point.room_name)
+            room_id = get_id_of("rooms", point.room_name, building_id=building_id)
 
         # {'room_id': room_id}
-        device_id = get_id_of("devices", point.device_name)
+        point.device_name = point.device_name or "UnID'd Device in " + point.building_name \
+                            + " " + point.room_name
+        device_id = get_id_of("devices", point.device_name, room_id=room_id)
```

Comment:
Is there a reason you don't need to know the building in this case too? Or is that taken care of in the device_name

Reply:
Same question goes for line 75.